addons/web/static/src/core/domain.js

```diff
@@ -206,6 +206,9 @@ function matchCondition(record, condition) {
             return fieldValue === value;
         case "!=":
         case "<>":
+            if (Array.isArray(fieldValue) && Array.isArray(value)) {
+                return !shallowEqual(fieldValue, value);
+            }
```

Comment:
Can you provide the input values for which this function wasn't working as you expect? I guess one of the value if `false`? I'm not sure this function should handle such a case, as false isn't a valid x2many values (the value of an unset x2many is the empty array).

Reply:
Hi @aab-odoo 
This is the domain thats not working:
https://github.com/odoo/odoo/blob/master/addons/payment/views/account_invoice_views.xml#L13
`('authorized_transaction_ids', '!=', [])`

Reply:
Ok but what was the value of `authorized_transaction_ids` in your case? Or could you give me the steps to reproduce your issue on a runbot?

Reply:
https://github.com/odoo/odoo/pull/96614